### PR TITLE
Change kernel from notebook

### LINKED ISSUE #752: Change kernel
Changing the kernel doesn't work.

## Description
When trying to change the kernel, there is no response.
The [Kernels class](https://github.com/jupyterlite/jupyterlite/blob/c5f87fc0dcf3dd597d819c50296b71bd427dd201/packages/kernel/src/kernels.ts#L20) has only `startNew`, `restart`, and `stop`. Maybe it is necessary to implement more functionalities, but I'm not sure, I haven't debugged this issue in depth yet.

## Reproduce
Open a notebook, and try to change the kernel.
For example, open the p5.js example and try to change the kernel to the `javascript`. It doesn't work.

Instead, open the same notebook and change the kernel to `No kernel` and then change it to `javascript`. It does work.

## Context
https://github.com/QuantStack/jupyterlab-blockly/issues/55

https://user-images.githubusercontent.com/26092748/181744193-faebf13c-2d6c-4324-9063-3ce3ae6f46cd.mp4

### PR DESCRIPTION
**Pull request by hbcarlos**
## References
Fixes #752 

## Code changes
Implemented the logic to change the kernel of a session.

## Backwards-incompatible changes
N/A

https://user-images.githubusercontent.com/26092748/182847438-0edd44c7-c951-4d9d-a98e-db1a09b40c33.mp4

### CODE REVIEW

File: packages/session/src/sessions.ts [hunk 1]
```diff
@@ -62,6 +65,29 @@ export class Sessions implements ISessions {
       path: path ?? session.path,
       name: name ?? session.name,
     };
+
+    if (kernel) {
+      // Kernel id takes precedence over name.
+      if (kernel?.id) {
```
Comment (jtpio) [inline packages/session/src/sessions.ts:71]: Looks like the `?` here might not be needed if there is already a guard above with `if (kernel)`?

File: packages/session/src/sessions.ts [hunk 2]
```diff
@@ -62,6 +65,29 @@ export class Sessions implements ISessions {
...
+        const session = this._sessions.find(
+          (session) => session.kernel?.id === kernel?.id
+        );
+        if (session) {
+          patched['kernel'] = session?.kernel;
```
Comment (jtpio) [inline packages/session/src/sessions.ts:76]: Wondering if it's possible to use the following notation?

```suggestion
          patched.kernel = session.kernel;
```

Normally TypeScript should be able to infer the type based on the function return type.

**Comment by hbcarlos** [comment]
Thanks Jeremy

File: packages/session/src/sessions.ts [hunk 3]
```diff
@@ -62,6 +65,29 @@ export class Sessions implements ISessions {
...
+        }
+      } else if (kernel.name) {
+        const newKernel = await this._kernels.startNew({
+          id: UUID.uuid4(),
+          name: kernel.name,
+          location: PathExt.dirname(patched.path),
+        });
+
+        if (newKernel) {
+          patched['kernel'] = newKernel;
```
Comment (jtpio) [inline packages/session/src/sessions.ts:86]: Another one here: 

```suggestion
          patched.kernel = newKernel;
```

**Review by jtpio — APPROVED** [review]
Thanks!

**Pull request merged**